spec/support/dummy_app_setup.rb

```diff
@@ -4,19 +4,29 @@ module DummyAppSetup
   module_function
 
   def setup
-    ENV["CONFIG_FILE_PATH"] = "#{CommandHelpers.spec_directory}/dummy/.controlplane/controlplane.yml"
+    config_dir = "#{CommandHelpers.spec_directory}/dummy/.controlplane"
+
+    config_file = File.read("#{config_dir}/controlplane.yml")
+    config_file = config_file.gsub("{GLOBAL_IDENTIFIER}", CommandHelpers.dummy_test_app_global_identifier)
+
+    @tmp_config_file = Tempfile.create(["controlplane-tmp-", ".yml"], config_dir)
+    @tmp_config_file.write(config_file)
+    @tmp_config_file.rewind
+
+    ENV["CONFIG_FILE_PATH"] = @tmp_config_file.path
```

Comment:
We need to read the original `controlplane.yml` file, replace the `{GLOBAL_IDENTIFIER}` placeholder, and save it to a new temporary file in the same directory.